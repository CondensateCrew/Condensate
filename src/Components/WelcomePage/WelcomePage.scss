@import "../../css/index.scss";

.welcome-page {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  min-height: 95vh;
  align-items: center;
  &>* {
    width: 50%;
  }
  aside {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    flex-direction: column;
    align-items: right;
    h2 {
      font-weight: 300;
      line-height: 1.5em;
      letter-spacing: 0.5em;
      text-align: right;
      margin-top: 4rem;
      margin-right: 1.25rem;
    }
  }
  &>div {
    @extend %initial-form;
    input:read-only:focus {
      outline: 0;
      box-shadow: $inner-shadow;
    }
  }
  &>.bubble-loading {
    transform: scale(3);
  }
}
