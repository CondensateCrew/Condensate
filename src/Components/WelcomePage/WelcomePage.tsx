import React, { useState } from 'react';
import './WelcomePage.scss';
import LoginForm from '../LoginForm/LoginForm';
import SignUpForm from '../../Containers/SignUpForm/SignUpForm';
import LoginForm from '../../Components/LoginForm/LoginForm';

const WelcomePage: React.FC = () => {
  const [ isLogin, setIsLogin ] = useState<boolean>(false);

  return (
    <main className="welcome-page">
      <aside>
        <h1>Condensate</h1>
        <h2>a new way <br /> to brainstorm</h2>
      </aside>
      { !isLogin ? <SignUpForm isLogin={isLogin} toggleTab={setIsLogin} />
      : <LoginForm isLogin={isLogin} toggleTab={setIsLogin}/>}
    </main>
  );
}

export default WelcomePage;
