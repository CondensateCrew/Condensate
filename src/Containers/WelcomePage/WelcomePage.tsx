import './WelcomePage.scss';
import React, { useState, useEffect } from 'react';
import { useCookies } from 'react-cookie';
import { useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import LoginForm from 'Components/LoginForm/LoginForm';
import SignUpForm from 'Components/SignUpForm/SignUpForm';
import LoadingImage from 'Components/LoadingImage/LoadingImage';
import { getSetUp, getDashboard } from 'apiCalls/apiCalls';
import { addUser, addWordSamples, addAllActions, addAllBrainstorms, addAllCategories, addSecretSauce } from 'redux/actions';
import { WordSample } from 'interfaces';

const WelcomePage: React.FC = () => {
  const [ isLoaded, setIsLoaded ] = useState<boolean>(false);
  const [ isLoading, setIsLoading ] = useState<boolean>(false);
  const [ isLogin, setIsLogin ] = useState<boolean>(true);
  const [ token, setToken ] = useState<string>('');
  const [ name, setName ] = useState<string>('');
  const [ error, setError ] = useState<string>('');
  const [ cookies, setCookie ] = useCookies(['token', 'name']);
  const dispatch = useDispatch();

  useEffect(() => {
    if (cookies.token) setToken(cookies.token);
    if (cookies.name) setName(cookies.name);
  }, [ cookies ]);

  useEffect(() => {
    if (name !== '' && token !== '') {
      const nameParts: string[] = name.split('-');
      dispatch(addUser({
        id: token,
        firstName: nameParts[0],
        lastName: nameParts[1]
      }));
    }
  }, [ dispatch, name, token ]);

  const fetchGameStuff = async (token: string) => {
    try {
      const setUpRes = await getSetUp(token);
      dispatch(addWordSamples(setUpRes));
      const { actions, categories, brainstorms } = await getDashboard(token);
      dispatch(addAllActions(actions));
      dispatch(addAllCategories(categories));
      dispatch(addAllBrainstorms(brainstorms));
      const secretSauce = await setUpRes
        .map((word: WordSample): string => word.word);
      dispatch(addSecretSauce(secretSauce));
      setIsLoaded(true);
    }
    catch (error) {
      setError(error.message);
    }
  };

  useEffect(() => {
    if (token !== '') fetchGameStuff(token);
  }, [ token ]); //eslint-disable-line

  const formSelected: JSX.Element = (isLogin)
    ? <LoginForm
      {...{isLogin, error, setCookie, isLoading, setIsLoading}}
      toggleTab={setIsLogin}/>
    : <SignUpForm
      {...{isLogin, error, setCookie, isLoading, setIsLoading}}
      toggleTab={setIsLogin} />

  return (
    <main className="welcome-page">
      { isLoaded && <Redirect to='/dashboard' />}
      <aside>
        <h1>Condensate</h1>
        <h2>a new way <br /> to brainstorm</h2>
      </aside>
      { token !== '' ? <LoadingImage /> : formSelected }
    </main>
  );
}

export default WelcomePage;
