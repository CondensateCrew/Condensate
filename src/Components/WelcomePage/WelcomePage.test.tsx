/**
 * @jest-environment jsdom
 */
import React from 'react';
import WelcomePage from './WelcomePage';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}));

describe('WelcomePage Component', () => {
  let wrapper;

  beforeEach(() => {
    jest.mock('react-router-dom', () => ({
      useHistory: () => ({
        push: jest.fn(),
      }),
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot with isLogin set to true and display LoginForm', () => {
    wrapper = shallow(<WelcomePage />)

    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot with isLogin set to false and display SignUpForm', () => {
    wrapper = mount(<MemoryRouter initialEntries={[ { pathname: '/', key: 'testKey' } ]}><WelcomePage /></MemoryRouter>)

    wrapper.find('.hidden').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
