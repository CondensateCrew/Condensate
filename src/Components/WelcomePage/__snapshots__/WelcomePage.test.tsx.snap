// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage Component should match the snapshot with isLogin set to false and display SignUpForm 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "culmuf",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "culmuf",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <WelcomePage>
      <main
        className="welcome-page"
      >
        <aside>
          <h1>
            Condensate
          </h1>
          <h2>
            a new way 
            <br />
             to brainstorm
          </h2>
        </aside>
        <SignUpForm
          isLogin={false}
          toggleTab={[Function]}
        >
          <div
            className="sign-up-form"
          >
            <header>
              <h2
                className="hidden"
                onClick={[Function]}
              >
                Login
              </h2>
              <h2
                className="active"
              >
                Sign Up
              </h2>
            </header>
            <form>
              <InputElement
                key="first-name"
                repeatPassword=""
                setError={[Function]}
                setPassword={[Function]}
                setUser={[Function]}
                typeInput="first-name"
                user={
                  Object {
                    "email": "",
                    "firstName": "",
                    "lastName": "",
                    "password": "",
                  }
                }
              >
                <label
                  htmlFor="first-name"
                >
                  first name
                  *
                  <input
                    id="first-name"
                    onChange={[Function]}
                    placeholder="Your First Name"
                    type="text"
                    value=""
                  />
                </label>
              </InputElement>
              <InputElement
                key="last-name"
                repeatPassword=""
                setError={[Function]}
                setPassword={[Function]}
                setUser={[Function]}
                typeInput="last-name"
                user={
                  Object {
                    "email": "",
                    "firstName": "",
                    "lastName": "",
                    "password": "",
                  }
                }
              >
                <label
                  htmlFor="last-name"
                >
                  last name
                  *
                  <input
                    id="last-name"
                    onChange={[Function]}
                    placeholder="Your First Name"
                    type="text"
                    value=""
                  />
                </label>
              </InputElement>
              <InputElement
                key="email"
                repeatPassword=""
                setError={[Function]}
                setPassword={[Function]}
                setUser={[Function]}
                typeInput="email"
                user={
                  Object {
                    "email": "",
                    "firstName": "",
                    "lastName": "",
                    "password": "",
                  }
                }
              >
                <label
                  htmlFor="email"
                >
                  email
                  *
                  <input
                    id="email"
                    onChange={[Function]}
                    placeholder="email@mail.com"
                    type="text"
                    value=""
                  />
                </label>
              </InputElement>
              <InputElement
                key="password"
                repeatPassword=""
                setError={[Function]}
                setPassword={[Function]}
                setUser={[Function]}
                typeInput="password"
                user={
                  Object {
                    "email": "",
                    "firstName": "",
                    "lastName": "",
                    "password": "",
                  }
                }
              >
                <label
                  htmlFor="password"
                >
                  password
                  *
                  <input
                    id="password"
                    onChange={[Function]}
                    placeholder="Your Password"
                    type="password"
                    value=""
                  />
                </label>
              </InputElement>
              <InputElement
                key="repeat-password"
                repeatPassword=""
                setError={[Function]}
                setPassword={[Function]}
                setUser={[Function]}
                typeInput="repeat-password"
                user={
                  Object {
                    "email": "",
                    "firstName": "",
                    "lastName": "",
                    "password": "",
                  }
                }
              >
                <label
                  htmlFor="repeat-password"
                >
                  repeat password
                  *
                  <input
                    id="repeat-password"
                    onChange={[Function]}
                    placeholder="Repeat Password"
                    type="password"
                    value=""
                  />
                </label>
              </InputElement>
              <button
                disabled={true}
                onClick={[Function]}
              >
                Sign Up
              </button>
            </form>
          </div>
        </SignUpForm>
      </main>
    </WelcomePage>
  </Router>
</MemoryRouter>
`;

exports[`WelcomePage Component should match the snapshot with isLogin set to true and display LoginForm 1`] = `
<main
  className="welcome-page"
>
  <aside>
    <h1>
      Condensate
    </h1>
    <h2>
      a new way 
      <br />
       to brainstorm
    </h2>
  </aside>
  <LoginForm
    isLogin={true}
    toggleTab={[Function]}
  />
</main>
`;
